### PR TITLE
Fix the revision suffix validation bug

### PR DESCRIPTION
**Pull request by ruslany**
---

### Description
Fix the revision suffix validation bug for az containerapp create command. The bug prevented using revision suffixes that started with numbers.

### Related command
export BLUE_COMMIT_ID=0b699ef
az containerapp create --name $APP_NAME \
  --environment $APP_ENVIRONMENT_NAME \
  --resource-group $RESOURCE_GROUP \
  --image mcr.microsoft.com/k8se/samples/test-app:$BLUE_COMMIT_ID \
  --revision-suffix $BLUE_COMMIT_ID \
  --env-vars REVISION_COMMIT_ID=$BLUE_COMMIT_ID \
  --ingress external \
  --target-port 80 \
  --revisions-mode multiple

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by ahmelsayed — APPROVED** [review]

**Comment by ruslany** [comment]
> Hi @ruslany, Please write the description of changes which can be perceived by customers into HISTORY.rst. If you want to release a new extension version, please update the version in setup.py as well.

fixed.

**Comment by ruslany** [comment]
Issue #6449

File: src/containerapp/azext_containerapp/_validators.py [hunk 1]
```diff
@@ -41,9 +41,13 @@ def _is_number(s):
 
 def validate_revision_suffix(value):
     if value is not None:
-        matched = re.match(r"^[a-z](?!.*-{2})([-a-z0-9]*[a-z0-9])?$", value)
+        # what does the following regex check?
+        # 1. ^[a-z0-9] - starts with a letter or number
+        # 2. (?!.*-{2}) - does not contain '--'
+        # 3. ([-a-z0-9]*[a-z0-9])? - ends with a letter or number and can contain '-' in between
+        matched = re.match(r"^[a-z0-9](?!.*-{2})([-a-z0-9]*[a-z0-9])?$", value)
         if not matched:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_validators.py:49]: Could you please add some test about this validate?

Comment (ruslany) [reply src/containerapp/azext_containerapp/_validators.py:49]: Added

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@ruslany Please take a look at this comment https://github.com/Azure/azure-cli-extensions/pull/6448#issuecomment-1610269407

**Comment by ruslany** [comment]
> @ruslany Please take a look at this comment [#6448 (comment)](https://github.com/Azure/azure-cli-extensions/pull/6448#issuecomment-1610269407)

@zhoxing-ms - thanks! Updated.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**